src/lib/bootloader/stage1_proposal.rb

```diff
@@ -95,6 +95,14 @@ def propose
 
     private
 
+      def any_boot_flag_partition?(disk_name)
+        disks = devicegraph.disks.with(name: disk_name)
+        legacy_boot = disks.first.partition_table.partition_legacy_boot_flag_supported?
```

Comment:
Are you sure the disk will always have a partition table at this point? If not this can blow away